dm/worker/subtask.go

```diff
@@ -73,9 +73,13 @@ type SubTask struct {
 	l log.Logger
 
 	sync.RWMutex
-	wg     sync.WaitGroup
+	wg sync.WaitGroup
+	// ctx is used for the whole subtask. It will be created only when we new a subtask.
 	ctx    context.Context
 	cancel context.CancelFunc
+	// currCtx is used for one loop. It will be created each time we use st.run/st.Resume
+	currCtx    context.Context
+	currCancel context.CancelFunc
```

Comment:
Can we do not keep these two fields in the `struct` (to make it clear)?

Reply:
Do you mean use one context?

Reply:
pass the local variable between methods (it seems not enough?)

Reply:
How to pass local variable between methods?